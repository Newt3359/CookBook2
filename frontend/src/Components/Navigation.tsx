import {useLocation, useNavigate} from 'react-router-dom'
import {BadgePlus, BookOpenText, ChefHat, Home} from "lucide-react";

export function Navigation(){

    const navigate = useNavigate();

    const location = useLocation();

    const currentPage = location.pathname

    const handleHome = () =>{
        navigate("/");
    }

    const handleAddRecipe = () =>{
        navigate("/add")
    }

    const handleReadRecipe = () =>{
        navigate("/read")
    }

    return(
        <>
            <nav className={"border-b-2 border-black bg-blue-300 text-black shadow-md sticky top-0 z-50"} data-testid={"navbar"}>
                <div className={"max-w-12xl mx-auto px-4 sm:px-6 lg:px-8"}>
                    <div className={"flex items-center justify-between h-20"}>
                        <div className={"flex items-center"}>
                            <ChefHat className={"size-20"}/>
                        </div>
                        <div className={"flex space-x-3 items-center"}>

                            <button
                                onClick={handleHome}
                                className={`flex items-center gap-2 px-4 py-2 rounded-full transition-all ${
                                    currentPage === "/"
                                        ? "bg-orange-200 text-black shadow-md"
                                        : "text-[#403F4C] hover:bg-orange-300/30"
                                }`}
                            >
                                <Home className="w-4 h-4" />
                                <span>Home</span>
                            </button>

                            <button
                                onClick={handleAddRecipe}
                                className={`flex items-center gap-2 px-4 py-2 rounded-full transition-all ${
                                    currentPage === "/add"
                                        ? "bg-orange-200 text-black shadow-md"
                                        : "text-[#403F4C] hover:bg-orange-300/30"
                                }`}
                            >
                                <BadgePlus className={"w-4 h-4"}/>
                                <span>Add Recipe</span>
                            </button>

                            <button
                                onClick={handleReadRecipe}
                                className={`flex items-center gap-2 px-4 py-2 rounded-full transition-all ${
                                    currentPage === "/read"
                                        ? "bg-orange-200 text-black shadow-md"
                                        : "text-[#403F4C] hover:bg-orange-300/30"
                                }`}
                            >
                                <BookOpenText className="w-4 h-4" />
                                <span>Read Recipes</span>
                            </button>

                        </div>
                    </div>
                </div>
            </nav>
        </>
    )
}
