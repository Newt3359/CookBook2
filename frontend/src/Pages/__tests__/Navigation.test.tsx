import {render, screen} from "@testing-library/react";
import {expect, describe, it} from "vitest";
import '@testing-library/jest-dom'
import {MemoryRouter} from "react-router-dom";
import {Navigation} from "../../Components/Navigation.tsx";
import {ChefHat} from "lucide-react";
import renderer from 'react-test-renderer';


describe("Navigation Bar Tests", () => {

    it('should render', () => {
        render(<MemoryRouter><Navigation/></MemoryRouter>)
        expect(screen.getByTestId("navbar"))
    });

    it('should have chef hat', () => {
        render(<MemoryRouter><Navigation/></MemoryRouter>)
        const chef = renderer.create(<ChefHat/>).toJSON();
        expect(chef).toMatchSnapshot();
    });

    it('should have a home button', () => {
        render(<MemoryRouter><Navigation/></MemoryRouter>)
        expect(screen.getByRole("button", {name: "Home"})).toBeVisible()
    });

    it('should have a add recipe button', () => {
        render(<MemoryRouter><Navigation/></MemoryRouter>)
        expect(screen.getByRole("button", {name: "Add Recipe"})).toBeVisible()
    });

    it('should have a view recipes button', () => {
        render(<MemoryRouter><Navigation/></MemoryRouter>)
        expect(screen.getByRole("button", {name: "Read Recipes"})).toBeVisible()
    });
})