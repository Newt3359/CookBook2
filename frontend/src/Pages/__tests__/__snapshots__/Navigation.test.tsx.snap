// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation Bar Tests > should have chef hat 1`] = `null`;
